pallets/file-system/src/tests.rs

```diff
@@ -366,6 +392,140 @@ fn bsp_volunteer_success() {
 }
 
+#[test]
+fn bsp_volunteer_storage_request_not_found_fail() {
+    new_test_ext().execute_with(|| {
+        let bsp_account_id = AccountId32::new([2; 32]);
+        let bsp_signed = RuntimeOrigin::signed(bsp_account_id.clone());
+        let location = FileLocation::<Test>::try_from(b"test".to_vec()).unwrap();
+        let fingerprint = H256::zero();
+        let multiaddr = BoundedVec::try_from(vec![1]).unwrap();
+        let multiaddresses: BoundedVec<
+            MultiAddress<Test>,
+            <Test as Config>::MaxDataServerMultiAddresses,
+        > = BoundedVec::try_from(vec![multiaddr]).unwrap();
+
+        assert_ok!(Providers::bsp_sign_up(
+            bsp_signed.clone(),
+            100,
+            multiaddresses.clone(),
+        ));
+
+        assert_noop!(
+            FileSystem::bsp_volunteer(
+                bsp_signed.clone(),
+                location.clone(),
+                fingerprint,
+                multiaddresses.clone()
+            ),
+            Error::<Test>::StorageRequestNotFound
+        );
+    });
+}
+
+#[test]
+fn bsp_already_volunteered_failed() {
+    new_test_ext().execute_with(|| {
+        let owner_account_id = AccountId32::new([1; 32]);
+        let owner_signed = RuntimeOrigin::signed(owner_account_id.clone());
+        let bsp_account_id = AccountId32::new([2; 32]);
+        let bsp_signed = RuntimeOrigin::signed(bsp_account_id.clone());
+        let location = FileLocation::<Test>::try_from(b"test".to_vec()).unwrap();
+        let size = 4;
+        let file_content = b"test".to_vec();
+        let fingerprint = BlakeTwo256::hash(&file_content);
+        let multiaddr = BoundedVec::try_from(vec![1]).unwrap();
+        let multiaddresses: BoundedVec<
+            MultiAddress<Test>,
+            <Test as Config>::MaxDataServerMultiAddresses,
+        > = BoundedVec::try_from(vec![multiaddr]).unwrap();
+        let storage_amount: StorageData<Test> = 100;
+
+        // Dispatch storage request.
+        assert_ok!(FileSystem::issue_storage_request(
+            owner_signed.clone(),
+            location.clone(),
+            fingerprint,
+            size,
+            multiaddresses.clone(),
+        ));
+
+        // Sign up account as a Backup Storage Provider
+        assert_ok!(Providers::bsp_sign_up(
+            bsp_signed.clone(),
+            storage_amount,
+            multiaddresses.clone(),
+        ));
+
+        // Dispatch BSP volunteer.
+        assert_ok!(FileSystem::bsp_volunteer(
+            bsp_signed.clone(),
+            location.clone(),
+            fingerprint,
+            multiaddresses.clone()
+        ));
+
+        assert_noop!(
+            FileSystem::bsp_volunteer(
+                bsp_signed.clone(),
+                location.clone(),
+                fingerprint,
+                multiaddresses.clone()
+            ),
+            Error::<Test>::BspAlreadyVolunteered
+        );
+    });
+}
+
+#[test]
+fn bsp_volunteer_threshold_too_high_fail() {
```

Comment:
The name of the error `ThresholdTooHigh` is a bit confusing. How about `AboveThreshold`?